### DISCUSSION TITLE
File upload loading indicator stopping after upload url signed and not upload finished

### DISCUSSION THREAD

**Discussion by MarllonGomes** [livewire/livewire#6814] [7. Bugs (archived)]
### Livewire version

3.0.2

### Which PHP version are you using?

PHP 8.2

### Steps To Reproduce

I have noticed that the loading indicator in this Livewire component stops right after the upload URL gets signed, and not after the actual file gets uploaded. I didn't notice until some slow mobile internet users reported that validation was not passing (validation not included here).
Now i reproduced with chrome network throthling at fast 3g option.

This is my livewire Class
```
<?php

namespace App\Livewire\Front;

use Livewire\Component;
use Livewire\WithFileUploads;

class FormWrapper extends Component
{
    use WithFileUploads;

    public $photo;

    public function render()
    {
        return view('livewire.front.form-wrapper');
    }
}
```

The Blade View:
```
<div>
    <input type="file" wire:model="photo">
    <div wire:loading wire:target="photo">Uploading...</div>
</div>
```

### Additional information

The project has FilamentPHP installed (v3.0.39), but the component isn't placed in a FilamentPHP panel; it is in a frontend view.

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by AdamEsterle** [comment]
I have the same problem. In Livewire 2, my "Save" button would stay disabled the entire time it was uploading to the temp dir
Now it flashes disable/enable very quickly at the beginning, stays enabled during upload, and then flashes again (disable to enabled) when uploading is finished.

Only after this second flash is the file actually on the server and can be saved (in my case sent to S3)

**Comment by robsontenorio** [comment]
@MarllonGomes  @AdamEsterle  Did you guys found any workaround or is it a bug?

**Reply by AdamEsterle** [reply]
It's a bug, but this is a workaround

```
{{--                Livewire 3 broke the nice button disabled, so we are showing this instead--}}
<div
    x-data="{ uploading: false, progress: 0 }"
    x-on:livewire-upload-start.window="uploading = true"
    x-on:livewire-upload-finish.window="uploading = false"
    x-on:livewire-upload-error.window="uploading = false"
    x-on:livewire-upload-progress.window="progress = $event.detail.progress"
    x-text="`Uploading, please wait. ${progress}%`"
    x-show="uploading"
>
</div>
```

It's really not that hacky, but annoying. I'm sure you can tie in the upload boolean to the disabled attr on the button

The clean and simple
`wire:loading.attr="disabled"`
no longer works
